Dockerfile

```
@@ -0,0 +1,89 @@
+# syntax=docker/dockerfile:1
+
+ARG CUDA_VER=11.8.0
+ARG PYTHON_VER=3.10
+ARG LINUX_VER=ubuntu22.04
+
+ARG RAPIDS_VER=23.06
+ARG DASK_SQL_VER=2023.2.0
+
+ARG BASE_FROM_IMAGE=rapidsai/mambaforge-cuda
+
+# Gather dependency information
+FROM rapidsai/ci:latest AS dependencies
+ARG CUDA_VER
+ARG PYTHON_VER
+
+ARG RAPIDS_VER
+
+ARG RAPIDS_BRANCH="branch-${RAPIDS_VER}"
+
+RUN pip install --upgrade conda-merge rapids-dependency-file-generator
+
+COPY notebooks.sh /notebooks.sh
+
+RUN /notebooks.sh
+
+
+# Base image
+FROM ${BASE_FROM_IMAGE}:cuda${CUDA_VER}-base-${LINUX_VER}-py${PYTHON_VER} as base
+ARG CUDA_VER
+ARG PYTHON_VER
+
+ARG RAPIDS_VER
+ARG DASK_SQL_VER
+
+RUN useradd -rm -d /home/rapids -s /bin/bash -g conda -u 1001 rapids
+
+USER rapids
+
+WORKDIR /home/rapids
+
+COPY condarc /opt/conda/.condarc
+
+COPY entrypoint.sh /home/rapids/entrypoint.sh
+
+# CI should handle modifying this file instead of the dockerfile
+# RUN if [ "${RAPIDS_BRANCH}" = "main" ]; then sed -i '/nightly/d;/dask\/label\/dev/d' /opt/conda/.condarc; fi
+
+RUN mamba install -y -n base \
+        "rapids=${RAPIDS_VER}.*" \
+        "dask-sql=${DASK_SQL_VER%.*}.*" \
+        "python=${PYTHON_VER}.*" \
+        "cudatoolkit=${CUDA_VER%.*}.*" \
```

Would suggest switching to the `cuda-version` package. This properly [constrains `cudatoolkit` on CUDA 11]( https://github.com/conda-forge/cuda-version-feedstock/blob/7e8b642c6bb4aeda93938a4cedc13895388937fe/recipe/meta.yaml#L17 ) and [on CUDA 12 this constrains the CTK]( https://github.com/conda-forge/cuda-version-feedstock/tree/main/recipe#readme )

```suggestion
        "cuda-version=${CUDA_VER%.*}.*" \
```

---

I think we _also_ need to install `cudatoolkit` on CUDA 11?

---

Am suggesting the latter, but have no strong preference